### PR TITLE
T15075

### PR DESCRIPTION
**Pull request by smspillaz**
https://phabricator.endlessm.com/T15075

### CODE REVIEW

File: data/application.css [hunk 1]
```diff
@@ -244,3 +248,36 @@ row:selected:backdrop.contact-row {
...
+  background-position: 100% 25%;
+  border-top-right-radius: 5px;
+}
+
+.thumbnail:hover .magnifying-glass-overlay {
+  background-image: url('resource:////com/endlessm/Coding/Chatbox/img/over_zoom.png');
+}
+
+.close-button {
+  background-image: url('resource:///com/endlessm/Coding/Chatbox/img/close_button.svg');
```
Comment (cosimoc) [inline data/application.css]: I am afraid this file is not part of the PR? But why use a custom icon for this, instead of the regular one coming from the theme?

File: src/views.js [hunk 2]
```diff
@@ -354,7 +358,19 @@ function getPreviewForFile(path, thumbnailFactory) {
...
-        if (thumbnailPath && GLib.file_test(thumbnailPath, GLib.FileTest.EXISTS)) {
+        // We need to check that both the thumbnail exists and that the
+        // mtime of the thumbnail is greater than the mtime of the original
+        // file
+        let mtimeThumbnail = (Gio.File.new_for_path(thumbnailPath)).query_info([
+            Gio.FILE_ATTRIBUTE_TIME_MODIFIED
+        ].join(','), Gio.FileQueryInfoFlags.NONE, null).get_modification_time();
+
+        let mTimeUsecs = timevalToUsecs(mtime);
+        let mTimeThumbnailUsecs = timevalToUsecs(mtimeThumbnail);
```
Comment (cosimoc) [inline src/views.js]: Instead of all of this, you should just check the `G_FILE_ATTRIBUTE_THUMBNAIL_IS_VALID` attribute, which does the same (and possibly more checks) internally

File: data/application.css (hunk 1 above)
Comment (smspillaz) [inline data/application.css]: *selfslap*

We were using the one coming from the theme but this asset came through per request of the design team.

File: src/views.js (hunk 2 above)
Comment (smspillaz) [inline src/views.js]: Looks handy, thanks!
